board: em_starterkit: Add 2.2 firmware version support

fanghuaqi opened the pull request:
Hi, all,

For EM Starter Kit, it has different firmware versions, even you can upgrade firmware 2.2 to firmware 2.3, but 2.2 EM7D and 2.3 EM7D have too many differences between each other.

For 2.2 EM7D, it has a normal MPU option, but 2.3 EM7D has secure MPU option, which are different, and currently, 2.3 EM7D is not supported. So we want to keep EM Starter Kit 2.2 and 2.3 supported together in Zephyr.

For supporting different firmware versions, `CONFIG_BOARD_EM_STARTERKIT_R22` and `CONFIG_BOARD_EM_STARTERKIT_R23` are introduced, you can change the `/boards/arc/em_starterkit/em_starterkit_defconfig` to select which firmware version you want to use.

Thanks
Huaqi

fanghuaqi committed:
arc: Remove unused backslash in makefile of arc em9d

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi committed:
boards: em_starterkit: Add board firmware version config option

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi committed:
arc: em7d: Add support for em starterkit 2.2 EM7D

Since em starterkit has different firmware versions(2.2 and 2.3),
but the EM7D of 2.3 has new secureshield feature, which is not supported
in Zephyr, but EM7D of 2.2 is a normal EM core, which can be supported,
so we add support for 2.2 EM7D.

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi committed:
arc: em7d: Clean up arch/arc/soc/em7d/soc.h

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi committed:
arc: em9d/em11d: Modify and cleanup soc.h

em starterkit has two versions, 2.2 and 2.3.
Change soc.h to support both versions,
main changes are the interrupt connections.

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi committed:
arc: em9d/em11d: Update default configurations

Update configurations to support em9d and em11d in
em starterkit 2.2 and 2.3

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi committed:
arc: em7d: Report error when build for 2.3's em7d

When you build application for em starterkit 2.3 em7d, it will
report error during build since it is not supported currently.

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi committed:
boards: em_starterkit: Move peripheral configs from soc to board level

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi committed:
doc: em_starterkit: add EM Starter Kit 2.2 support

Update EM Starter Kit documentation to cover support for
EM Starter Kit 2.2.

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

dbkinder commented on boards/arc/em_starterkit/doc/board.rst:
```diff
@@ -63,6 +63,22 @@ which have the EM7D, EM9D, or EM11D SoC.
+
+The EM Starter Kit has different versions, such as 1.0, 1.1, 2.0, 2.1,
+2.2 and 2.3.
+In Zephyr, 2.2 and 2.3 firmware versions are supported.
```
How about, 

     In Zephyr, only firmware versions 2.2 and 2.3 are supported.

fanghuaqi replied:
OK, thanks for suggestion, I will modify it.

dbkinder commented on boards/arc/em_starterkit/doc/board.rst:
```diff
@@ -63,6 +63,22 @@ which have the EM7D, EM9D, or EM11D SoC.
+In Zephyr, 2.2 and 2.3 firmware versions are supported.
+
+* For EM Starter Kit 2.2, EM7D, EM9D and EM11D core configurations are supported.
+  * Use `CONFIG_BOARD_EM_STARTERKIT_R22` to select 2.2 version.
```
you can get a hyperlink to the configuration documentation for these options by using

     :option:`CONFIG_BOARD_EM_STARTERKIT_R22`

and do the same for the references to config options below.

dbkinder commented on boards/arc/em_starterkit/doc/board.rst:
```diff
@@ -63,6 +63,22 @@ which have the EM7D, EM9D, or EM11D SoC.
+* For EM Starter Kit 2.2, EM7D, EM9D and EM11D core configurations are supported.
+  * Use `CONFIG_BOARD_EM_STARTERKIT_R22` to select 2.2 version.
+  * Use `CONFIG_SOC_EM7D`, `CONFIG_SOC_EM9D` or `CONFIG_SOC_EM11D` to select
+  EM7D, EM9D or EM11D.
```
add "respectively" before the period.

dbkinder commented on boards/arc/em_starterkit/doc/board.rst:
```diff
@@ -63,6 +63,22 @@ which have the EM7D, EM9D, or EM11D SoC.
+  EM7D, EM9D or EM11D.
+* For EM Starter Kit 2.3, EM9D and EM11D core configurations are supported.
+  * Use `CONFIG_BOARD_EM_STARTERKIT_R23` to select 2.3 version.
+  * Use `CONFIG_SOC_EM9D` or `CONFIG_SOC_EM11D` to select EM9D or EM11D.
```
same here, add "respectively" before the period.

dbkinder commented on boards/arc/em_starterkit/doc/board.rst:
```diff
@@ -151,6 +167,25 @@ setup the board correctly.
+the default EM Starter Kit board default configuration file
+:file:`/boards/arc/em_starterkit/em_starterkit_defconfig`.
+
+For example, if you choose EM Starter Kit 2.2 EM11D, you need to change the
```
since these line numbers may change, I'd recommend just saying, 

     For example, if you choose EM Starter Kit 2.2 EM11D, you need to change the
     ``em_starterkit_defconfig`` file to say:

dbkinder reviewed (changes requested):
some edits recommended

nashif reviewed (changes requested):
there is a merge commit in the series, please rebase the set against master and resubmit.

fanghuaqi committed:
doc: em_starterkit: updated according to pull request #1017

Make some edits according to the suggestions provided by #1017

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

fanghuaqi commented:
I have rebased the branch against master, and make some edits in the documentations, and update this branch, thanks

fanghuaqi committed:
doc: em_starterkit: Fix doc issue reported by shippable

See
https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/3090/3/tests
for details

Signed-off-by: Huaqi Fang <huaqi.fang@synopsys.com>

dbkinder reviewed (approved):
LGTM
